### ISSUE TITLE
Computed fields with all arguments with default values still require an arg object

### ISSUE THREAD

**Issue by Kaelten** [hasura/graphql-engine#8525]
### Version Information

Server Version: 2.6.0

### Environment

OSS

### What is the expected behaviour?

When a custom field is using a function that has all additional args with default values, the generated schema should make the args object optional.  

Enabling this creates a better story around backwards-compatibility when iterating over an existing resolver that was initially created without additional arguments. 

### Keywords

computed field  args

### What is the current behaviour?

The computed field requires an arg object even if all of the function arguments are optional.

### How to reproduce the issue?

1. Create a postgres function with the row argument and a second one that has a default value.
2. Attach function as computed field
3. attempt to use it without passing in the args object.

### Screenshots or Screencast

<img width="139" alt="image" src="https://user-images.githubusercontent.com/18547/170144484-53407b03-4099-458c-8cd8-ab793ff39bb6.png">

### If the bug is confirmed, would you be willing to submit a PR?

I do not have the knowledge needed to update a haskell codebase.

**Comment by rakeshkky** [comment]
Hi @Kaelten, I can see, from initial glance at the issue description, the query goes through if we specify `args: {}` as input to `related` field.

Triaging the issue now. I think, the fix lies in [`functionArgs`](https://github.com/hasura/graphql-engine/blob/d5cb312f8b3bdd8f86c210ad5b7e26903a6fc596/server/src-lib/Hasura/GraphQL/Schema/Select.hs#L1442) function.

**Comment by Kaelten** [comment]
> 

Correct.  I saw the same thing in my experimentation.  However (best I can tell) with the current implementation it becomes disruptive to add optional args to a function that's been previously released without one.
